pandas/tests/tslibs/test_period_asfreq.py

```diff
@@ -1,10 +1,32 @@
 # -*- coding: utf-8 -*-
 
+import pytest
+
+from pandas.errors import OutOfBoundsDatetime
 from pandas._libs.tslibs.frequencies import get_freq
-from pandas._libs.tslibs.period import period_ordinal, period_asfreq
+from pandas._libs.tslibs.period import period_ordinal, period_asfreq, Period
 
 
 class TestPeriodFreqConversion(object):
+    @pytest.mark.parametrize('freq', ['A', 'Q', 'M', 'W', 'B', 'D'])
+    def test_asfreq_near_zero(self, freq):
+        # GH#19643, GH#19650
+        per = Period('0001-01-01', freq=freq)
+        tup1 = (per.year, per.hour, per.day)
+
+        prev = per - 1
+        assert (per - 1).ordinal == per.ordinal - 1
```

Comment:
i think as a general rule if you are importing Period, then it should go in pandas/tests/scalar/period/* else it can go here (e.g. you are testing a helper function)

Reply:
Seeing as how the recent addition of tests.scalar.period will necessitate a new look at this, I propose we merge this bug fix and the next pass (later today) can attempt to standardize this convention.

Reply:
I think you can also simply move the tests there now (as travis has to run again anyhow since you just added a commit)

Reply:
can you update for this comment